client/deposits/details/index.tsx

```diff
@@ -142,16 +129,85 @@ export const DepositOverview: React.FC< DepositOverviewProps > = ( {
 				)
 			}
 			value={ <DepositStatusIndicator deposit={ deposit } /> }
-			detail={ deposit.bankAccount }
+			detail={
+				<>
+					{ deposit.bankAccount }
+					<br />
+					Bank reference ID:{ ' ' }
+					{ deposit.bank_reference_key ? (
+						<>
+							<span className="woopayments-payout-bank-reference-id">
+								{ deposit.bank_reference_key }
+							</span>
+							<button
+								type="button"
+								className={ `woopayments-payout-bank-reference-id-copy-button ${
+									isCopied ? 'state--copied' : ''
+								}` }
+								aria-label={ __(
+									'Copy bank reference ID to clipboard',
+									'woocommerce-payments'
+								) }
+								title={ __(
+									'Copy to clipboard',
+									'woocommerce-payments'
+								) }
+								onClick={ () => {
+									const bankReferenceId = document.querySelector(
+										'.woopayments-payout-bank-reference-id'
+									)?.textContent;
```

Comment:
How can we get the value without fishing around in the DOM (clue: `deposit` prop 😁)? We try to avoid that in React; the component should not depend on the DOM, it should be self contained.

Reply:
Awesome hint! I wonder how I missed it 😄 

Tweaked it here - [af176bd](https://github.com/Automattic/woocommerce-payments/pull/9886/commits/af176bd9a1f5bcb52689c0f9d8b13ded513b792f)